CoreScriptsRoot/ServerCoreScripts/ServerChat/ChatService.lua

```diff
@@ -0,0 +1,285 @@
+local source = [[
+local module = {}
+
+local modulesFolder = script.Parent
+
+--////////////////////////////// Include
+--//////////////////////////////////////
+local ChatChannel = require(modulesFolder:WaitForChild("ChatChannel"))
+local Speaker = require(modulesFolder:WaitForChild("Speaker"))
+
+--////////////////////////////// Details
+--//////////////////////////////////////
+local metatable = {}
+metatable.__ClassName = "ChatService"
+
+metatable.__tostring = function(tbl)
+	return tbl.__ClassName .. ": " .. tbl.MemoryLocation
+end
+
+metatable.__metatable = "The metatable is locked"
+metatable.__index = function(tbl, index, value)
+	if rawget(tbl, index) then return rawget(tbl, index) end
+	if rawget(metatable, index) then return rawget(metatable, index) end
+	error(index .. " is not a valid member of " .. tbl.__ClassName)
+end
+metatable.__newindex = function(tbl, index, value)
+	error(index .. " is not a valid member of " .. tbl.__ClassName)
+end
+
+
+--////////////////////////////// Methods
+--//////////////////////////////////////
+function metatable:Dump()
+	return tostring(self)
+end
+
+function metatable:DoMessageFilter(speakerName, message)
+	for i, func in pairs(self.FilterMessageFunctions) do
+		pcall(function()
+			local ret = func(speakerName, message)
+			assert(type(ret) == "string")
+			message = ret
+		end)
+	end
+	
+	message = self.FilterMessageFunctions["default_filter"](speakerName, message)
+	
+	if (game:GetService("RunService"):IsServer() and not game:GetService("RunService"):IsStudio()) then
```

Comment:
Feels like we should get the RunService once and put it in a variable instead of multiple times